lib/fetchers/Permit2Fetcher.ts

```diff
@@ -31,7 +42,9 @@ export class Permit2Fetcher {
       const beforePermitCheck = Date.now();
       const rpcProvider = this.chainIdRpcMap.get(chainId);
       if (!rpcProvider) throw new Error(`No rpc provider found for chain: ${chainId}`);
-      allowance = await this.permit2.connect(rpcProvider).allowance(ownerAddress, tokenAddress, spenderAddress);
+      const permit2 = this.chainIdPermit2Map.get(chainId);
```

Comment:
i still dont understand why we need this map built into this class? cant this line just be
`const permit2 = permit2Address(chainId)`
and fetch it from the import? why does it need to reconstruct that function locally in `chainIdPermit2Map`?

Reply:
Note `this.permit2` is a `ethers.Contract` instance, not a vanilla contract address. `ethers.Contract` instance needs to be a singleton instance here, because we don't want to re-instantiate `ethers.Contract` instance per chain, per request. Ideally we only have a single `ethers.Contract` singleton instance, but in this PR, we have to have two `ethers.Contract` singleton instances, one for zksync chain, and the other for the rest of the supported chains.

Note in the `ethers.Contract` constructor, there are some heavy lifting process:
-  RPC call to resolve name (https://github.com/ethers-io/ethers.js/blob/v5.7/packages/contracts/src.ts/index.ts#L717)
- decode function signatures (https://github.com/ethers-io/ethers.js/blob/v5.7/packages/contracts/src.ts/index.ts#L734)
- decode events (https://github.com/ethers-io/ethers.js/blob/v5.7/packages/contracts/src.ts/index.ts#L686)

We don't want to have those repeated per request.